extensions/Push/lib/Connector/Webhook.pm

```diff
@@ -0,0 +1,199 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::Push::Connector::Webhook;
+
+use 5.10.1;
+use strict;
+use warnings;
+
+use base 'Bugzilla::Extension::Push::Connector::Base';
+
+use Bugzilla;
+use Bugzilla::Bug;
+use Bugzilla::Constants;
+use Bugzilla::Attachment;
+use Bugzilla::Extension::Webhooks::Webhook;
+use Bugzilla::Extension::Push::Constants;
+use Bugzilla::Extension::Push::Util;
+use Bugzilla::Util ();
+
+use JSON qw(decode_json encode_json);
+use LWP::UserAgent;
+use List::MoreUtils qw(any);
+
+sub options {
+  return (
+    {
+      name     => 'id',
+      label    => 'Webhook id',
+      type     => 'string',
+      default  => '',
+      required => 1,
+    },
+    {
+      name     => 'url',
+      label    => 'URL',
+      type     => 'string',
+      default  => '',
+      required => 1,
+    },
+    {
+      name     => 'name',
+      label    => 'Name',
+      type     => 'string',
+      default  => '',
+      required => 1,
+    },
+    {
+      name     => 'event',
+      label    => 'Event',
+      type     => 'string',
+      default  => '',
+      required => 1,
+    },
+    {
+      name     => 'product_name',
+      label    => 'Product name',
+      type     => 'string',
+      default  => '',
+      required => 1,
+    },
+    {
+      name     => 'component_name',
+      label    => 'Component name',
+      type     => 'string',
+      default  => '',
+      required => 0,
+    },
+  );
+}
+
+sub new {
+  my ($class,$webhook_id) = @_;
+  my $self = {};
+  bless($self, $class);
+  ($self->{name}) = 'Webhook_' . $webhook_id;
```

Comment:
I think we can't do this, because the connector is initialized with the webhook_id, from there get the object webhook to configure it (in the load_config) so if we don't pass it we don't have the self->id, and in the name is named with that when is created and as isn't still configure when is named, it don't have the self->id so we need to pass it too.